File: UsersForums/UsersForumClient/users-forum-app/src/app/models/response.ts
import { Post } from './post';
import{ User } from './user';

export class Response {
  id: number;
  postId: number;  // Related post
  userId: number;  // User who posted the response
  name: string; // his name
  responseMessage?: string;
isEditing?: boolean;
  // Navigation Properties
  //post: Post;  // Related post
  //user: User;  // Responder user

  constructor(
    id: number,
    postId: number,
    userId: number,
    name:string,
    //post: Post,
   // user: User,
    responseMessage?: string
  ) {
    this.id = id;
    this.postId = postId;
    this.userId = userId;
    this.name = name;
    // this.post = post;
    //this.user = user;
    this.responseMessage = responseMessage;
  }
}

File: UsersForum/UsersForumService/UsersForumService.DAL/Entities/Response.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace UsersForumService.DAL.Entities
{
    public class Response
    {
        public int Id { get; set; }
        public int PostId { get; set; }  // Related post
        public int UserId { get; set; }  // User who posted the response
        public string? Name { get; set; } // Resposer Name
        public string? ResponseMessage { get; set; }

        // Navigation Properties
        [JsonIgnore] 
        public Post? Post { get; set; }  // Related post
     
        [JsonIgnore] 
        public User? User { get; set; }  // Responder user
    }
}
